zerver/lib/bugdown/fenced_code.py

```diff
@@ -125,136 +125,141 @@ def extendMarkdown(self, md: markdown.Markdown, md_globals: Dict[str, Any]) -> N
                              position)
 
 
+class BaseHandler:
```

Comment:
So this piece I don't like about this commit is mostly that `fenced_code` is very slightly forked version of the upstream https://python-markdown.github.io/extensions/fenced_code_blocks/, and so it kinda sucks to do this restructuring from the standpoint of being able to integrate with upstream changes in the future.

How much impact does this piece have on bugdown perf?  Might be worth it anyway.

Reply:
As we discussed on chat, this module is already heavily customized, and the perf improvement is measurable, and the code change is probably better for readability in the long run.  (It's the codehilite module that we don't want to re-vendor.)